lib/src/io.dart

```diff
@@ -216,6 +215,10 @@ Future<String> _createFileFromStream(Stream<List<int>> stream, String file) {
   });
 }
 
+void _chmod(int mode, String file) {
+  runProcessSync('chmod', [mode.toRadixString(8), file]);
```

Comment:
We could perhaps also use `dart:ffi` to bind the syscall... But we a little care has to be taken that we don't bind symbols only exposed because the Linux build is in release mode and not product mode.

Reply:
It seems to be `/bin/chmod` on macOS too, so that should work.

Using `dart:ffi` sounds interesting (especially because we could check the value of `umask` at runtime too). AFAIK Dart doesn't support linking libc statically, so the symbols should be available